### PR TITLE
Convert to java8

### PR DESCRIPTION
**Pull request by myronkscott**

### CODE REVIEW

File: packaging-support/src/main/java/com/tc/classloader/PermanentEntity.java [hunk 1]
```diff
@@ -30,8 +31,10 @@
  */
 @Target( ElementType.TYPE )
 @Retention( RetentionPolicy.RUNTIME )
+@Repeatable( Permanents.class )
 public @interface PermanentEntity {
   String type();
-  String[] names();
+  String[] names() default {};
```
Comment (ramsai1729) [inline packaging-support/src/main/java/com/tc/classloader/PermanentEntity.java]: what is the use of names() now?

File: packaging-support/src/main/java/com/tc/classloader/OverrideService.java [hunk 2]
```diff
@@ -19,20 +19,22 @@
...
 
 /**
  * Tagging a Service type with an override causes the classloader to ignore the 
- * overridden types for the type provided.  Each type is the fully qualified classname
- * of the type to be overridden.  The overridden types the and type of the interface 
- * tagged should be the same from the perspective of the classloader.
+ * overridden types for the type provided.  The value is a fully qualified class name 
+ * which is to be overridden.
  * 
  * TODO: consider using the @Repeatable once java 6 support is dropped
```
Comment (ramsai1729) [inline packaging-support/src/main/java/com/tc/classloader/OverrideService.java]: this comment can be removed now.

File: packaging-support/src/main/java/com/tc/classloader/PermanentEntity.java (hunk 1 above)
Comment (myronkscott) [reply packaging-support/src/main/java/com/tc/classloader/PermanentEntity.java]: Backward compatible.  You can now do either 

@PermanentEntity(type=type,names=[one,two])
public class EntityServerService

or

@PermanentEntity(type=type,names=one)
@PermanentEntity(type=type,names=two)
public class EntityServerService

File: packaging-support/src/main/java/com/tc/classloader/OverrideService.java (hunk 2 above)
Comment (myronkscott) [reply packaging-support/src/main/java/com/tc/classloader/OverrideService.java]: updated

**Pull request merged**